src/components/Button/Button.scss

```diff
@@ -124,6 +124,11 @@
   border: none;
   vertical-align: top;
   line-height: normal;
+  font-size: 0;
+
+  > * + * {
```

Comment:
I think the options are either `* + *` which covers every scenario including elements non-standard elements that someone might want to add or a big selector such as `span + span, .ms-Button-icon + .ms-Button-description, .ms-Button-description + .ms-Button-icon, .ms-Button-icon + .ms-Button-text, .ms-Button-text + .ms-Button-icon...` 

I agree `* + *` is a horrible selector